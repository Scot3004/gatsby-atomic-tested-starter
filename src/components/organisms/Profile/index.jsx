import React from 'react';
import PropTypes from 'prop-types';
import MainTitle from '../../atoms/MainTitle';
import Avatar from '../../atoms/Avatar';
import ProfileCard from '../../atoms/ProfileCard';

export const Profile = ({ name, avatar }) => {
  return (
    <ProfileCard>
      <Avatar src={avatar} alt={name} size="96px" />
      <MainTitle>{name}</MainTitle>
    </ProfileCard>
  );
};

Profile.propTypes = {
  name: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired
};

export default Profile;
