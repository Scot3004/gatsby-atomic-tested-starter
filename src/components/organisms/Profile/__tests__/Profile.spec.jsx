import React from 'react';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import Profile from '..';

describe('When I visit the site', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Profile name="sergio" avatar="img_stub" />);
  test('Then the profile of the blog owner should be shown', () => {
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
