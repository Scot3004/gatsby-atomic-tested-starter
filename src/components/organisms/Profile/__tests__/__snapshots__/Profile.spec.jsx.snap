// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I visit the site Then the profile of the blog owner should be shown 1`] = `
<ForwardRef>
  <ForwardRef
    alt="sergio"
    size="96px"
    src="img_stub"
  />
  <ForwardRef>
    sergio
  </ForwardRef>
</ForwardRef>
`;
